rows.go

```diff
@@ -262,8 +262,10 @@ func (r *rows) Close() error {
 	C.duckdb_destroy_result(&r.res)
 
 	if r.stmt != nil {
-		C.duckdb_destroy_prepare(r.stmt.stmt)
 		r.stmt.rows = false
+		if r.stmt.closeOnRowsClose {
+			r.stmt.Close()
```

Comment:
there is no explicit error being returned from `Close()` method on `statement`.  Except not sure if error should be actionable, may be just log the error as potential stmt leak ? @begelundmuller what are your thoughts here.

Reply:
Fixed. Even though it can't currently return an error, I think it's a good practice to propagate errors in case the func starts returning errors in the future.